### ISSUE TITLE
ENG: Update buttons and links in profile area

### ISSUE THREAD

**Issue by brea11y** [department-of-veterans-affairs/va-mobile-app#7814]

**Comment by theodur** [comment]
@brea11y In order to change the role of the buttons on the Contact information and Settings screens to 'menuitem', the accessibility role for the `BaseListItem` component (which is the foundations for all other list components in the app) needed to be updated. This means items in all the other lists like: 
- `DefaultList` (e.g. the lists on the Contact information screen)
- `MessageList` (e.g. the list of messages on the Inbox screen)
- `SimpleList` (e.g. the lists on the Messages folders screen)
- `PickerList` (used in the VAModalPicker component for the picker items)

will have `menuitem` as their accessibility role. Is the`menuitem` role okay for all other lists in the app as well? The role for items in these lists is currently `button`.

**Comment by theodur** [comment]
@brea11y Also wanted to confirm something, for the `Profile / Contact information / Remove [information] alert` bullet point, are you referring to the button that's on top of the edit screens for removing the items? e.g. the `Remove home phone` button on the Home phone screen? If so, all of these buttons are already destructive, so no changes are needed there

**Comment by theodur** [comment]
@brea11y For the `Open settings` bullet point, would adding an accessibility hint to the button with the text `Opens system settings` be good enough to let the user know the button will launch them out of the app? So the button would be announced `Open settings, button. Opens system settings`. Or should this be an actual alert pop-up that gives the user the options to proceed or cancel?

**Comment by brea11y** [comment]
> @brea11y In order to change the role of the buttons on the Contact information and Settings screens to 'menuitem', the accessibility role for the `BaseListItem` component (which is the foundations for all other list components in the app) needed to be updated. This means items in all the other lists like:
> 
>     * `DefaultList` (e.g. the lists on the Contact information screen)
> 
>     * `MessageList` (e.g. the list of messages on the Inbox screen)
> 
>     * `SimpleList` (e.g. the lists on the Messages folders screen)
> 
>     * `PickerList` (used in the VAModalPicker component for the picker items)
> 
> 
> will have `menuitem` as their accessibility role. Is the`menuitem` role okay for all other lists in the app as well? The role for items in these lists is currently `button`.

Yes. Menu item would be fine. Since they are used for navigation and not for an action, menu item would be a better fit if link is not an option.

**Comment by brea11y** [comment]
> @brea11y Also wanted to confirm something, for the `Profile / Contact information / Remove [information] alert` bullet point, are you referring to the button that's on top of the edit screens for removing the items? e.g. the `Remove home phone` button on the Home phone screen? If so, all of these buttons are already destructive, so no changes are needed there

No, this was referencing to the native alerts that launch after you tap "Remove home phone", for example. There is a native alert that appears after that with options to "Keep" or "Remove". "Remove". We've updated the alert style since this audit and don't have the option to make the "Remove" link red anymore, so I will remove this AC from the list.

**Comment by brea11y** [comment]
> @brea11y For the `Open settings` bullet point, would adding an accessibility hint to the button with the text `Opens system settings` be good enough to let the user know the button will launch them out of the app? So the button would be announced `Open settings, button. Opens system settings`. Or should this be an actual alert pop-up that gives the user the options to proceed or cancel?

Thanks for checking. It would be better if this was the actual native alert that warns a user that they will be leaving the app. We can use the same "Leave the mobile app?" alert that we use throughout the app. At the time of the audit, we had not aligned this alert to be the same throughout the app, so I will update the AC to remove the last bit about opening their device settings.

**Comment by theodur** [comment]
@brea11y Thanks for confirming! For reference, the standard alert that's used when tapping on an external link in the app has verbiage that's more specific to launching a website via the browser, i.e. `You'll go to a website outside of this app.`. I had to change the body text a little to be suited to opening the system settings, so the body for the alert says `You'll go to the system settings for this app.`

<details>
<summary> Screenshot </summary> 

![Screenshot_1720715753](https://github.com/department-of-veterans-affairs/va-mobile-app/assets/9042882/d08476e9-cba7-4525-b37b-7b4bbb23f947)

</details>

**Comment by TKDickson** [comment]
Blocked by lack of access to testing build; adding blocker ticket & label.

**Comment by TKDickson** [comment]
OK unblocked via process, but over my WIP limit, adding blocker label back again

**Comment by TKDickson** [comment]
picking this up today

**Comment by TKDickson** [comment]
Found #9209 during testing; caused months ago so written up as a separate ticket.

**Comment by TKDickson** [comment]
@theodur on my OnePlus I'm seeing the thing with disability rating again where it announces the entire box, and then you can swipe through each row in the box (when it should be JUST the box). And in addition, the 'each row' bit is announcing that it's interactive (see screenshot). Not happening on iOS/that's behaving normally.
<img width="365" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOVVZQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--44963eddda80be16496fcfd35a016d571f8f908e/image.png" alt="image.png" />

**Comment by TKDickson** [comment]
Found #9120, also not caused by this development.

**Comment by TKDickson** [comment]
Moving back to in-progress for the issue a couple of comments above.

**Comment by theodur** [comment]
@TKDickson Weird that error is back 🤔, I remember resolving that a while ago. The changes I made wouldn't have impacted that screen. I just confirmed that issue is also happening in the latest `develop`. We'll probably need a new bug ticket for that

**Comment by TKDickson** [comment]
Wrote up that comment as #9236. Which means this ticket is good to move forward.

**Comment by TKDickson** [comment]
I just realized I should add the list of testing. In addition to explicitly checking ACs, I also checked lists (actionable and non-actionable), links, and buttons on all of these places, iOS and Android:
* Claims file upload
* Claim status
* Notifications
* Remote config
* Enable debug
* Developer screen
* Privacy policy
* Manage account
* contact info
* personal info
* Date range pickers (payment, appt)
* Other pickers (file type, messaging folder)
* Claim letters (& every other 'feature that is a list' I could think of)
* Letters list
* BSSV letter screen
* Disability rating
* Refill workflow (selection list)
* Preferred name
* Gender Identity
* claims ask for decision
